subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractConsoleFunctionalSpec.groovy

```diff
@@ -46,6 +46,24 @@ class AbstractConsoleFunctionalSpec extends AbstractIntegrationSpec {
         return styledString
     }
 
+    /**
+     * Grabs the grouped output of a task
+     * @param taskName a fully qualified task name e.g. :project:task1
+     * @return
+     */
+    protected String taskOutputFrom(String taskName) {
+        // Starting with >, including the task name, and going till the first new line
+        def taskHeader = "> Task $taskName[^\\n]*\n"
+        // Start of the next task
+        def taskFooter = "\n\n[^\\n]*> Task"
+        // End of the build, or the build failure message
+        def buildFooter = "\n\n\n[^\\n]*(BUILD SUCCESSFUL|BUILD FAILED|FAILURE:)"
+        //Find all output between the header and footer
+        def matcher = result.output =~ /(?msU)\$taskHeader(.*?)(\$taskFooter|\$buildFooter)/
+
+        return matcher[0][1]
```

Comment:
I think we can do the first part with something like 
```groovy
assert matcher.find() : 'Task was not found in output'
```
I'm not sure I understand what the second part is... doesn't this basic test act as a defacto test of this method?